src/promise-btn.directive.ts

```diff
@@ -33,12 +33,21 @@ export class PromiseBtnDirective implements OnDestroy, AfterContentInit {
   }
 
   @Input()
-  set promiseBtn(promise: any) {
-    if (promise instanceof Observable) {
-      this.promise = promise.toPromise();
-    } else {
-      this.promise = promise;
+  set promiseBtn(request: any) {
+    const isObservable: boolean = request instanceof Observable;
+    const isPromise: boolean = request instanceof Promise || (
+      request !== null &&
+      typeof request === 'object' &&
+      typeof request.then === 'function' &&
+      typeof request.catch === 'function'
+    );
+
+    if (isObservable) {
+      this.promise = request.toPromise();
+    } else if (isPromise) {
+      this.promise = request;
```

Comment:
if passed value is not Observable and is not Promise, `this.promise` won't be set. It allows us to simplify checks below:
` if (!this.promise || !this.promise.then) {`
becomes
` if (!this.promise) {`